### ISSUE TITLE
ERROR! splitext() takes exactly 1 argument (0 given)

### ISSUE THREAD

**Issue by basictheprogram** [ansible/ansible#25771]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

VarsModule 'host_group_vars'

##### ANSIBLE VERSION

```
$ ansible --version
ansible 2.4.0 (devel 3eea649cbb) last updated 2017/06/15 21:40:19 (GMT -500)
  config file = None
  configured module search path = [u'/Users/tanner/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/tanner/projects/ansible.git/ansible/lib/ansible
  executable location = /Users/tanner/projects/ansible.git/ansible/bin/ansible
  python version = 2.7.10 (default, Feb  7 2017, 00:08:15) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)]
```

##### CONFIGURATION

ANSIBLE_DEBUG=1

##### OS / ENVIRONMENT

Ansible from macOS 10.12.5
Managed host Windows 10

##### SUMMARY

Today's git pull results in the error, yesterday's git pull worked as expected. 
Playbook unchanged from yesterday.

##### STEPS TO REPRODUCE

```yaml
TASK [Gathering Facts] *********************************************************
Not really a playbook
```

##### EXPECTED RESULTS

Ansible would run to completion

##### ACTUAL RESULTS

```
TASK [Gathering Facts] *********************************************************
ok: [PC18.local]
ERROR! splitext() takes exactly 1 argument (0 given)
```

**Comment by basictheprogram** [comment]
git bisect tell me this is the commit that broke things
```
commit 13a1111ddebbadf2db161724eb470154ecdccc88
Author: Brian Coca <brian.coca+git@gmail.com>
Date:   Thu Jun 15 12:34:05 2017 -0400

    tighter host/group_var discovery

    now only loads nonext or valid yaml extensions
    only first file gets loaded
    directories must match name (initially) so no name.yml is loaded anymore
```
And the culprit is lib/ansible/plugins/vars/host_group_vars.py:126

**Comment by bcoca** [comment]
fixed by #25797
